### ISSUE TITLE
Making API less PIMS-specific

### ISSUE THREAD

**Issue by danielballan** [soft-matter/slicerator#2]
Writing out the README, it bothered me that it takes so much code to write a minimal example.

Rough thoughts on an alternative:

`Slicerator` should take a function as its argument. The function will be expected to accept an integer as its argument. Then the minimal example looks like this:

```
def get_frame(i):
    ...

Slicerator(get_frame)
```

For richer applications, as in PIMS, where `get_frame` is a method of some object:

```
Slicerator.from_obj(ImageSequence, 'get_frame')
```

The second argument could be optional; it could default to `__getitem__`.

**Comment by danielballan** [comment]
```
class ImageSequence:

    __getitem__(self, val):
        return Slicerator.from_obj(self, 'get_frame', attrs_whitelist)
```

**Comment by danielballan** [comment]
Closed by #5
